packages/react-components/react-drawer/src/components/InlineDrawer/useInlineDrawer.ts

```diff
@@ -14,22 +14,20 @@ import { useDrawerDefaultProps } from '../../shared/useDrawerDefaultProps';
  * @param props - props from this instance of InlineDrawer
  * @param ref - reference to root HTMLElement of InlineDrawer
  */
-export const useInlineDrawer_unstable = (
-  props: InlineDrawerProps,
-  ref: React.Ref<HTMLDivElement>,
-): InlineDrawerState => {
+export const useInlineDrawer_unstable = (props: InlineDrawerProps, ref: React.Ref<HTMLElement>): InlineDrawerState => {
   const { size, position, open } = useDrawerDefaultProps(props);
   const { separator = false } = props;
 
-  const motion = useMotion<HTMLDivElement>(open);
+  const motion = useMotion<HTMLElement>(open);
+  const defaultTag = props.as || 'div';
 
   return {
     components: {
-      root: 'div',
+      root: defaultTag,
```

Comment:
This might not be necessary, as `slot.always` will respect the precedence of `props.as` instead of `components.root`